### PR TITLE
Add GHA and testing infrastructure

### PR DESCRIPTION
**Pull request by MrCreosote**
Also license, update readme with basics, etc.

### CODE REVIEW

File: .github/dependabot.yml
Comment (MrCreosote) [inline .github/dependabot.yml:1]: Standard GHA build and push scripts used everywhere

**Comment by MrCreosote** [comment]
Test are failing due to Codecov, sigh. It looks like you have to merge to the default branch at least once before they'll work now...?

https://app.codecov.io/gh/kbase/cdm-task-service/new

**Review by Tianhao-Gu — APPROVED** [review]
LGTM

**Comment by MrCreosote** [comment]
I accidentally pushed the next pr onto this PR, and so force pushed back to the original commit

**Pull request merged**